packages/studioqt/widgets/folderswidget/folderswidget.py

```diff
@@ -752,27 +752,28 @@ def columnCount(self, *args):
 
     def ignoreFilter(self):
         """
-        :rtype: list or None
+        :rtype: func or None
         """
         return self._ignoreFilter
 
     def setIgnoreFilter(self, ignoreFilter):
         """
-        :type ignoreFilter: list or None
+        :type ignoreFilter: func or None
         """
-        self._ignoreFilter = ignoreFilter or []
+        self._ignoreFilter = ignoreFilter
 
     def isPathValid(self, path):
         """
+        Return True if provided path is valid to show in widget
+
         :type path: str
         :rtype: bool
         """
-        if os.path.isdir(path):
-            path = path.lower()
-            valid = [item for item in self._ignoreFilter if path.endswith(item)]
-            if not valid:
-                return True
-        return False
+        if self._ignoreFilter:
```

Comment:
The folder widget should only show directories, so "os.path.isdir(path)" should be called even if there is no filter.

Example:

```python
   def isPathValid(self, path):
        valid = os.path.isdir(path)
        
        if valid and self._filter:
            valid = self._filter(path)

        return valid